lib/chef/provider/package.rb

```diff
@@ -304,6 +305,14 @@ def unlock_package(name, version)
         raise( Chef::Exceptions::UnsupportedAction, "#{self} does not support :unlock" )
       end
 
+      def validate_package(name)
+        raise( Chef::Exceptions::Package, "Package names containing whitespace characters such as spaces, tabs, or newlines are not allowed." ) if invalid_name?(name)
+      end
+
+      def invalid_name?(package_name)
+        ( package_name.is_a?(Array) && package_name.any? { |name| name.index(/\s|\\t|\\n/) } ) || ( package_name.is_a?(String) && package_name.to_s.index(/\s|\\t|\\n/) )
```

Comment:
I think you can make this clearer by just doing `Array(package_name).any? {  .... }` to coerce single strings to Arrays of strings and then do the same validation?